Remotion/ObjectBinding/Core/BindableObject/PropertyReflector.cs

```diff
@@ -166,6 +166,9 @@ protected virtual Type GetUnderlyingType ()
 
     protected virtual Type GetItemType ()
     {
+      if (!typeof (IEnumerable).IsAssignableFrom (_propertyInfo.PropertyType))
+        return _propertyInfo.PropertyType;
```

Comment:
There are no tests specifically for this class, should I create a new one, or is this case already covered by existing tests?
@MichaelKetting

Reply:
ok it's already covered by existing tests.

Reply:
I thought I have to write a new test to check if
```csharp
if (!typeof (IEnumerable).IsAssignableFrom (_propertyInfo.PropertyType))
        return _propertyInfo.PropertyType;
```
 works correctly

Reply:
> Unrelated to the test: I noticed that GetItemType is called more often than necessary given the performance overhead involved. Please check this an make suggestions.

We could simply store the value of `GetItemType()` in a variable instead of calling the method over and over again

Reply:
no, a mock doesn't change things. The problem is that we explicity added a shortcut to test for an interface common to all the collection types and return if it's not a collection type. so, you can't really "see" the extra conditionin the code, which is a typical "problem" of a performance optimizaiton. Of course, if you could observe the added IF, this would likely mean that a bug was introduced.  Considering code coverage, a test for a property of type IEnumerable could be helpful. That way, we have both branches covered (the first and the last one). The first branch is hit when a non-collection type is used, the last one, when it's a sequence and in between, for actual supported collections.

For testing of protected members,  you'd just derive and make method exposed via a public helper (unless we just use `PrivateInvole`).